app/routes.py

```diff
@@ -1 +1,317 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint
+from os import abort
+from app import db
+from flask import Blueprint, jsonify, abort, make_response, request
+from app.models.task import Task
+from app.models.goal import Goal
+from sqlalchemy import asc, desc
+import datetime
+import requests
+import os
+from dotenv import load_dotenv
+
+load_dotenv()
+
+
+tasks_bp = Blueprint("tasks_bp", __name__, url_prefix = "/tasks")
+goals_bp = Blueprint("goals_bp", __name__, url_prefix = "/goals")
+
+#CREATE 
+#task post route
+@tasks_bp.route("", methods=["POST"])
+def create_one_task():
+    request_body = request.get_json()
+
+    if "title" not in request_body:
+        return {
+            "details": "Invalid data"
+        }, 400
+    if "description" not in request_body:
+        return {
+            "details": "Invalid data"
+        }, 400
+
+    if "completed_at" not in request_body:
+        completed_at = None
+    else:
+        completed_at = request_body["completed_at"]
+    
+    new_task = Task(
+                title=request_body["title"],
+                description=request_body["description"],
+                completed_at=completed_at)
+
+
+    db.session.add(new_task)
+    db.session.commit()
+
+    return {
+        "task": new_task.to_dict()
+    }, 201 
+
+#goal post route
+@goals_bp.route("", methods=["POST"])
+def create_one_goal():
+    request_body = request.get_json()
+
+    if "title" not in request_body:
+        return {
+            "details": "Invalid data"
+        }, 400
+    
+    new_goal = Goal(
+                title=request_body["title"])
+
+
+    db.session.add(new_goal)
+    db.session.commit()
+
+    return new_goal.to_dict(), 201 
+
+#READ
+#task get routes
+@tasks_bp.route("", methods=["GET"])
+def read_all_tasks():
+    param = request.args
+    if "sort" in param:
+        if param["sort"] == "asc":
+            tasks = Task.query.order_by(Task.title.asc())
+        elif param["sort"] == "desc":
+            tasks = Task.query.order_by(Task.title.desc())
+    else:
+        tasks = Task.query.all()
+
+    tasks_response = []
+    for task in tasks:
+        tasks_response.append(task.to_dict())
+
+    return jsonify(tasks_response), 200
+
+
+@tasks_bp.route("/<task_id>", methods=["GET"])
+def read_one_task(task_id):
+    chosen_task = get_task_or_abort(task_id)
+    
+    response = {"task": chosen_task.to_dict()}
+
+    if chosen_task.goal_id:
+        response ={ "task": {
+                "id": chosen_task.task_id,
+                "goal_id": chosen_task.goal_id,
+                "title": chosen_task.title,
+                "description": chosen_task.description,
+                "is_complete": bool(chosen_task.completed_at)
+                }
+            }
```

Comment:
Tip: Some of this is logic duplicated with `to_dict`.  

You can DRY it up by adding the `goal_id` to result of `to_dict`:

```suggestion
        response = { 
            "task": chosen_task | { "goal_id": chosen_task.goal_id }
        }
```